Assets/Python/RiseAndFall.py

```diff
@@ -1053,235 +1042,124 @@ def checkPlayerTurn(self, iGameTurn, iPlayer):
+					for city in utils.getCityList(iPlayer):
+						x = city.getX()
+						y = city.getY()
+						pPlot = gc.getMap().plot(x, y)
 
-						if not city.isWeLoveTheKingDay() and not city.isCapital() and (city.getX(), city.getY()) != Areas.getCapital(iPlayer):
-							if (gc.getPlayer(iPlayer).getNumCities() > 0): #this check is needed, otherwise game crashes
+						if not city.isWeLoveTheKingDay() and not city.isCapital() and (x, y) != Areas.getCapital(iPlayer):
+							if gc.getPlayer(iPlayer).getNumCities() > 0: #this check is needed, otherwise game crashes
 								capital = gc.getPlayer(iPlayer).getCapitalCity()
-								iDistance = utils.calculateDistance(city.getX(), city.getY(), capital.getX(), capital.getY())
-								if (iDistance > 3):											       
-							    
-									if (city.angryPopulation(0) > 0 or \
-									    city.healthRate(False, 0) < 0 or \
-									    city.getReligionBadHappiness() > 0 or \
-									    city.getLargestCityHappiness() < 0 or \
-									    city.getHurryAngerModifier() > 0 or \
-									    city.getNoMilitaryPercentAnger() > 0 or \
-									    city.getWarWearinessPercentAnger() > 0):
+								iDistance = utils.calculateDistance(x, y, capital.getX(), capital.getY())
+								if iDistance > 3:
+							
+									if city.angryPopulation(0) > 0 or \
+										city.healthRate(False, 0) < 0 or \
+										city.getReligionBadHappiness() > 0 or \
+										city.getLargestCityHappiness() < 0 or \
+										city.getHurryAngerModifier() > 0 or \
+										city.getNoMilitaryPercentAnger() > 0 or \
+										city.getWarWearinessPercentAnger() > 0:
 										cityList.append(city)
 										continue
 									
 									for iLoop in range(iNumTotalPlayers+1):
-										if (iLoop != iPlayer):
-											if (pCurrent.getCulture(iLoop) > 0):
+										if iLoop != iPlayer:
+											if pPlot.getCulture(iLoop) > 0:
 												cityList.append(city)
 												break
 
-					if (len(cityList)):
+					if cityList:
 						iNewCiv = iIndependent
 						iRndNum = gc.getGame().getSorenRandNum(2, 'random independent')
-						if (iRndNum % 2 == 0):
-							iNewCiv = iIndependent2  
-						if (iPlayer == iAztecs or \
-						    iPlayer == iInca or \
-						    iPlayer == iMaya or \
-						    iPlayer == iEthiopia or \
-						    iPlayer == iMali):
-							if (data.iCivsWithNationalism <= 0):
-								iNewCiv = iNative		    
-						splittingCity = cityList[gc.getGame().getSorenRandNum(len(cityList), 'random city')]
-						utils.cultureManager((splittingCity.getX(),splittingCity.getY()), 50, iNewCiv, iPlayer, False, True, True)
-						utils.flipUnitsInCityBefore((splittingCity.getX(),splittingCity.getY()), iNewCiv, iPlayer)			    
-						data.tTempFlippingCity = (splittingCity.getX(), splittingCity.getY())
-						utils.flipCity((splittingCity.getX(),splittingCity.getY()), 0, 0, iNewCiv, [iPlayer])   #by trade because by conquest may raze the city
+						if iRndNum == 1:

```

Comment:
Not that it matters (I think), but `iRandNum % 2 == 0` and `iRandNum == 1` are opposite cases.